examples/step-62/step-62.cc

```diff
@@ -0,0 +1,1467 @@
+                      }
+                  }
+
+                // We loop again over the degrees of freedom of the cells to
+                // calculate the system matrix. These loops are really quick
+                // because we have already calculated the stiffness and mass
+                // matrices, only the value of $\omega$ changes.
+                for (unsigned int i = 0; i < dofs_per_cell; ++i)
+                  {
+                    for (unsigned int j = 0; j < dofs_per_cell; ++j)
+                      {
+                        std::complex<double> matrix_sum = 0;
+                        matrix_sum += -std::pow(omega, 2) *
+                                      quadrature_data.mass_coefficient[i][j];
+                        matrix_sum +=
+                          quadrature_data.stiffness_coefficient[i][j];
+                        cell_matrix(i, j) += matrix_sum * quadrature_data.JxW;
+                      }
+                    cell_rhs(i) += quadrature_data.right_hand_side[i];
+                  }
+              }
+            cell->get_dof_indices(local_dof_indices);
+            constraints.distribute_local_to_global(cell_matrix,
+                                                   cell_rhs,
+                                                   local_dof_indices,
+                                                   system_matrix,
+                                                   system_rhs);
+          }
+      }
+
+    system_matrix.compress(VectorOperation::add);
+    system_rhs.compress(VectorOperation::add);
+  }
+
+  // @sect4{ElasticWave::solve}
+
+  // This is even more simple than in step-40. We use the parallel direct solver
+  // MUMPS which requires less options than an iterative solver. The drawback is
+  // that it does not scale very well. It is not straightforward to solve the
+  // Helmholtz equation with an iterative solver. The shifted Laplacian
+  // multigrid method is a well known approach to precondition this system, but
+  // this is beyond the scope of this tutorial.
+  template <int dim>
+  void ElasticWave<dim>::solve()
+  {
+    TimerOutput::Scope              t(computing_timer, "solve");
+    LinearAlgebraPETSc::MPI::Vector completely_distributed_solution(
+      locally_owned_dofs, mpi_communicator);
+
+    SolverControl                    solver_control;
+    PETScWrappers::SparseDirectMUMPS solver(solver_control, mpi_communicator);
+    solver.solve(system_matrix, completely_distributed_solution, system_rhs);
+
+    pcout << "   Solved in " << solver_control.last_step() << " iterations."
+          << std::endl;
+    constraints.distribute(completely_distributed_solution);
+    locally_relevant_solution = completely_distributed_solution;
+  }
+
+  // @sect4{ElasticWave::set_position_vector}
+
+  // We use this function to calculate the values of the position vector.
+  template <int dim>
+  void ElasticWave<dim>::set_position_vector()
```

Comment:
I don't recall whether I commented on this before or not: a function that starts with `set_` generally (by convention, not necessity) takes an argument and sets a member variable. I think that you use this function *initialize* something, right? Maybe call it `initialize_position_vector()`.